backend-rust/src/migrations/m0002-block-cumulative-fin-time-index.sql

```diff
@@ -3,3 +3,6 @@
 CREATE INDEX blocks_height_null_cumulative_finalization_time ON blocks (height)
     WHERE blocks.cumulative_finalization_time IS NULL
     AND blocks.finalization_time IS NOT NULL;
+-- Used to efficiently perform partial string matching on the `hash` column,
+-- allowing fast lookups when searching for blocks by their hash prefix using `LIKE`.
+CREATE INDEX blocks_hash_gin_trgm_idx ON blocks USING gin(hash gin_trgm_ops);
```

Comment:
gin seems a bit heavy for just a prefix search, but it is probably okay.
Mostly concerned about the indexing time here.